sdcm/nemesis.py

```diff
@@ -2972,28 +2972,55 @@ def _write_read_data_to_multi_dc_keyspace(self, datacenters: List[str]) -> None:
                     f"-mode cql3 native compression=lz4 -rate threads=5 -pop seq=1..10000 -log interval=5"
         write_thread = self.tester.run_stress_thread(stress_cmd=write_cmd, use_single_loader=True)
         self.tester.verify_stress_thread(cs_thread_pool=write_thread)
-        read_cmd = f"cassandra-stress read no-warmup cl=ALL n=10000 -schema 'keyspace=keyspace_new_dc " \
-                   f"replication(strategy=NetworkTopologyStrategy,{datacenters[0]}=3,{datacenters[1]}=1) " \
+        self._verify_multi_dc_keyspace_data(consistency_level="ALL")
+
+    def _verify_multi_dc_keyspace_data(self, consistency_level: str = "ALL"):
+        read_cmd = f"cassandra-stress read no-warmup cl={consistency_level} n=10000 -schema 'keyspace=keyspace_new_dc " \
                    f"compression=LZ4Compressor' -port jmx=6868 -mode cql3 native compression=lz4 -rate threads=5 " \
                    f"-pop seq=1..10000 -log interval=5"
         read_thread = self.tester.run_stress_thread(stress_cmd=read_cmd, use_single_loader=True)
         self.tester.verify_stress_thread(cs_thread_pool=read_thread)
 
+    def _switch_to_network_replication_strategy(self, keyspaces: List[str]) -> None:
+        """Switches replication strategy to NetworkTopology for given keyspaces.
+        """
+        node = self.cluster.nodes[0]
+        nodes_by_region = self.tester.db_cluster.nodes_by_region()
+        region = list(nodes_by_region.keys())[0]
+        dc_name = self.tester.db_cluster.get_nodetool_info(nodes_by_region[region][0])['Data Center']
+        for keyspace in keyspaces:
+            replication_strategy = ReplicationStrategy.get(node, keyspace)
+            if not isinstance(replication_strategy, SimpleReplicationStrategy):
+                # no need to switch as already is NetworkTopology
+                continue
+            self.log.info(f"Switching replication strategy to Network for '{keyspace}' keyspace")
+            if keyspace == "system_auth":
+                # system_auth keyspace must have full replication
+                replication_strategy.replication_factor = len(nodes_by_region[region])
+            network_replication = NetworkTopologyReplicationStrategy(
+                **{dc_name: replication_strategy.replication_factor})
+            network_replication.apply(node, keyspace)
+            for node in nodes_by_region[0]:
+                node.run_nodetool(sub_cmd=f"repair {keyspace}", publish_event=True)
```

Comment:
I wonder.
You set it so that it would replicate data only into first dc.
At the same time you run it for `system_auth`.
In theory it means that authentication data wont be synced into second DC nodes.
Which potentially means that nodes of the seconds DC won't be able to authenticate connections

Reply:
This part is only preparation step for adding new DC (all system keyspaces must have network topology).
In next part of the nemesis, all system keyspaces get new dc node to rf factor (see lines 3018-3023).